src/v/raft/replicate_batcher.cc

```diff
@@ -33,6 +33,14 @@ replicate_stages replicate_batcher::replicate(
   std::optional<model::term_id> expected_term,
   chunked_vector<model::record_batch> batches,
   replicate_options opts) {
+    if (opts.as) [[unlikely]] {
```

Comment:
Should the unlikely be in abort_requested condition?

Reply:
I've put it here as our hot path doesn't use replicate abort source, so i wanted to give that branch a handicap with `unlikely`